### DISCUSSION TITLE
Waveform-data split by time ,or split arraybuffer?

### DISCUSSION THREAD

**Discussion by jason30704** [bbc/peaks.js#540] [Q&A] [answered]
I'm using MediaRecorder to record audio, and I want to real time show waveform during record.

As shown in the red box below. Complete audio can use waveform-data to obtain waveform data normally. But if I use the audio part of the Chunk, I will not be able to obtain the waveform data normally, as shown in the white box below.
![image](https://github.com/user-attachments/assets/2f1e1ecd-dd1d-4d3d-81d4-135734eadd00)

Below is the code I use to get waveform-data
![image](https://github.com/user-attachments/assets/b4a3f58b-b4e5-46b1-999c-50d013b2cefd)

The method in the red box will take more time to generate waveform-data as the recording time gets longer.
Is there a correct way to partially extract the waveform-data of an audio by waveform-data ?

**Comment by chrisn** [comment]
There's an existing issue: https://github.com/bbc/waveform-data.js/issues/86, so I'd suggest continuing the conversation there. It would would need changes to `WaveformData` to enable this.

**Comment by chrisn** [comment, marked as answer]
Acutally, there will be two parts to solving this: changing `WaveformData`, and then visualising using Peaks.js. See also https://github.com/bbc/peaks.js/issues/376.

**Reply by jason30704** [reply]
Thank you so much.
I think it can help me a lot. I'll give it a try.
